public/app/features/dashboard/components/PanelEditor/usePanelLatestData.ts

```diff
@@ -18,15 +24,16 @@ export const usePanelLatestData = (panel: PanelModel, options: GetDataOptions):
 
     return () => {
       if (querySubscription.current) {
-        console.log('unsubscribing');
         querySubscription.current.unsubscribe();
       }
     };
-  }, [panel]);
+  }, [panel, options]);
 
   return [
     latestData,
-    // TODO: make this more clever, use PanelData.state
-    !!(latestData && latestData.series),
+    // Loading state
+    latestData ? latestData.state === LoadingState.Loading : true,
+    // Error
+    latestData && latestData.error,
   ];
```

Comment:
Can we return an object here with typed properties instead of bools so it's easier to use this hook?

Reply:
Yes, good point!